rename_files.sh

```diff
@@ -1,76 +1,116 @@
+    echo "Give 3 command-line arguments. Or give 0 arguments & get prompts."
     exit 1
   fi
-  if [ ! "$2" ]; then
-    echo "Enter the string to replace as the 2nd parameter." 
+
+  if [ "$#" -eq 0 ]; then # Show prompts.
+    # Save old IFS value & restore it after last read command.
+    OLD_IFS=$IFS
+    IFS=''
+    
+    echo "Type the directory path: "
+    read -r directory_path
+  
+    echo "Type the old text: "
+    read -r old_text
+
+    echo "Type the new text: "
+    read -r new_text
+    IFS=$OLD_IFS
+  elif [ $# -lt 3 ]; then # Not enough command-line arguments given. Requires 3.
+    echo "$need_single_quotes"
+    echo "3 arguments are required."
+    if [ $# -lt 2 ]; then # Show message when user only gives 1 command-line argument.
+      echo "You are missing 2 arguments."
+    else
+      # Show message when user only gives 2 command-line arguments. 
+      echo "You are missing 1 argument."
+    fi
+    # Note: You must give '' empty string as 3rd argument if you want to replace 
+    # with nothing , i.e., remove text.
     exit 1
+  else # Gave required 3 command-line arguments.
+    directory_path="$1"
+    old_text="$2"
+    new_text="$3"
   fi
-  
-  # Note: If no 3rd argument given, then is interpreted as '' empty string.
-  # if [ ! "$3" ]; then
-  #  replacement_string=''
-  # fi
-  directory_path=$1
-  string_to_replace=$2
-  replacement_string=$3
-fi
+}
 
-echo "Directory path you typed: $directory_path"
-echo "String to replace you typed: $string_to_replace"
-echo "Replacement string you typed: $replacement_string"
-echo
+print_arguments() {
+  directory_path=${directory_path}
+
+  echo "Checking:"
+  echo "Directory path: $directory_path"
+  check_directory "$directory_path"
+
+  echo "Old text: $old_text"
+  echo "New text: $new_text"
+  echo
+}
 
-if [ ! -d "$directory_path" ] 
-then
+check_directory() {
+  if [ ! -d "$directory_path" ] 
+  then
     echo "This directory does NOT exist."
     exit 1
-fi
+  fi
+}
+
+need_single_quotes="If argument has SPACES, place argument within single quotes."
+check_for_arguments "$@"
+print_arguments
 
-echo "Filename change in progress..."
-echo "..."
+echo "Checking files..."
+file_sort_counter=0
+start=$(date +%s)
 
-find "$directory_path" -type f -name '*.odt' |
-while read a_file_name; do
+while read -r a_file_name; do
   # Replace only 1st match: ${string/pattern/replacement}
   # NOTE: Tricky since pattern must be in the specific format above & CANNOT be 
   # a variable.
+  # -e - flag for indicating the pattern you want to match against.
 
-  new_file_name="${a_file_name/$string_to_replace/$replacement_string}"
-  mv "$a_file_name" "$new_file_name"
-done
-echo "Done."
+  if (echo "$a_file_name" | grep -q -e "$old_text"); then
+    echo "FOUND $old_text IN:" "$(basename "$a_file_name")"
+    new_file_name="${a_file_name/$old_text/$new_text}"
+    mv "$a_file_name" "$new_file_name"
+    file_sort_counter="$((file_sort_counter+1))"
+  fi
```

Comment:
Although you have the echo command
```
  echo "FOUND $old_text IN:" "$(basename "$a_file_name")"
```
it's possible that the match for `$old_text` is actually in the directory part of `$a_file_name` and not in the base name part. Since you feed all of `$a_file_name` to `grep`, it could find a match before the last slash (`/`) instead of after it.

If you'd like to confine the matches to just the base name, you could instead send just the base name part of `$a_file_name` to `grep`:

```
if (echo "$(basename "$a_file_name")" | grep -q -e "$old_text"); then
  ...
fi
```

Reply:
@kcharter Good point. Added basename to condition.

Reply:
All fixed up now.